### DISCUSSION TITLE
Why does form state type returns TFormData directly?

### DISCUSSION THREAD

**Discussion by ha1fstack** [TanStack/form#756] [Q&A] [answered]
Form does not guarantee if the state matches the provided `TFormData` type, because the field values are mounted only when corresponding field component mounts.
Currently form state type returns the provided `TFormData` directly, so it's not typesafe and can cause runtime errors if not handled properly.
Shouldn't it be `DeepPartial<TFormData>`? Is there any context for this?

**Comment by crutchcorn** [comment, marked as answer]
If I'm understanding what you're saying properly, the state _is_ guaranteed as well as we can, provided you pipe us the correct data:

https://github.com/TanStack/form/issues/705#issuecomment-2135943113
